File: leetcode/Largest_Number.cpp
#include <iostream>
#include <vector>
#include <sstream>
using namespace std;


bool Greater(const int& a, const int& b)
{
    stringstream ssa, ssb;
    ssa << a << b;
    ssb << b << a;
    return ssa.str() > ssb.str();
}

class Solution {
public:
    string largestNumber(vector<int> &num) {

    	string res;

    	sort(num.begin(), num.end(), Greater);
        for(auto &a : num) res += to_string(a);

    	if(res[0] == '0') return "0";
    	return res;
    }
};


int main(int argc, char ** argv)
{
    Solution s;
    int as[] = {1125, 232};
    vector<int> vs(as, as + sizeof(as)/sizeof(int));

    cout << s.largestNumber(vs) << endl;
    return 0;
}

File: leetcode/Largest_Number.py





class Solution:
    # @param num, a list of integers
    # @return a string
    def largestNumber(self, num):
        num = sorted(num, cmp=self.bigger)
        res = ""
        for a in num: res = res + str(a)
        if res[0] is '0': return '0'
        return res

    def bigger(self, a, b):
        s1 = str(a) + str(b)
        s2 = str(b) + str(a)
        if s1<s2: return 1
        if s1>s2: return -1
        return 0


def main():
    num = [8, 123, 321, 92, 993, 9, 9]
    s = Solution()
    print s.largestNumber(num)

if __name__ == '__main__':
    main()
